kubetest/local.go

```diff
@@ -70,10 +71,39 @@ func (n localCluster) Up() error {
 	if err != nil {
 		return err
 	}
+
 	cmd := exec.Command(script)
 	cmd.Env = os.Environ()
 	cmd.Env = append(cmd.Env, "ENABLE_DAEMON=true")
 	cmd.Env = append(cmd.Env, fmt.Sprintf("LOG_DIR=%s", n.tempDir))
+
+	// when we are running in a DIND scenario, we should use the ip address of
+	// the docker0 network interface, This ensures that when the pods come up
+	// the health checks (for example for kubedns) succeed. If there is no
+	// docker0, just use the defaults in local-up-cluster.sh
+	dockerIp := ""
+	docker0, err := net.InterfaceByName("docker0")
+	if err == nil {
```

Comment:
bikeshed: maybe better to handle log errors here
```
if docker0, err := net.InterfaceByName("docker0"); err != nil {
  if addresses, err := docker0.Addrs(); err != nil {
    for _, address := range addresses {
      ...
    }
  } else {
    log.printf("error from docker0.Addrs() : %v", err)
  }
} else {
  log.printf("error from net.InterfaceByName : %v", err)
}

 I think, and you can print the error

Reply:
or, probably should return the err?

Reply:
@krzyzacy i'll just do the logging and fall back to the defaults in local-up-cluster.sh if we are unable to get info about docker0 net interface

Reply:
sgtm

Reply:
thanks updated.